File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Square.java
package edu.cmu.cs.cs214.lab02.shapes;

/**
 * Represents a square shape that implements the Shape interface.
 */
public class Square implements Shape {
    private final double sideLen; // 将字段设为 private 和 final

    /**
     * Constructs a Square with the specified side length.
     *
     * @param sideLen the length of the square's side
     */
    public Square(double sideLen) {
        this.sideLen = sideLen;
    }

    /**
     * Returns the area of the square.
     *
     * @return the area of the square
     */
    @Override
    public double getArea() {
        return sideLen * sideLen;
    }
}

File: typescript/src/shapes/square.ts
import type { Shape } from "./shape.js"; // 使用 type-only 导入

export class Square implements Shape {
    private readonly sideLen: number; // 将字段设为 private 和 readonly

    constructor(sideLen: number) {
        this.sideLen = sideLen;
    }

    getArea(): number {
        return this.sideLen * this.sideLen;
    }

    /**
     * Returns the side length of the square.
     *
     * @return the side length of the square
     */
    getSideLen(): number {
        return this.sideLen;
    }
}
